From 79bbada1d1a608fbfc55f16244ef0004bbcf1c05 Mon Sep 17 00:00:00 2001
From: Anil Dsouza <anil.dsouza@einfochips.com>
Date: Tue, 10 May 2022 15:40:28 +0530
Subject: [PATCH] media: i2c: ap1320: Auto focus actuator initialization

Initialization of the auto focus actuators is done from the driver code
instead of enabling it from the camera specific firmware binary file.

Change-Id: Id8f43a09c87d49c3459543e91f179d8c8d51fbfe
Signed-off-by: Anil Dsouza <anil.dsouza@einfochips.com>
---
 drivers/media/i2c/ap1302.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/drivers/media/i2c/ap1302.c b/drivers/media/i2c/ap1302.c
index 370bf0722..6e2fb087d 100644
--- a/drivers/media/i2c/ap1302.c
+++ b/drivers/media/i2c/ap1302.c
@@ -377,6 +377,9 @@
 #define AP1302_TCLK_PRE_SHIFT			0x8
 
 #define AP1302_ADV_MIPI_FIX			AP1302_REG_32BIT(0x00730000)
+#define ACT_ABS_MTIME				AP1302_REG_32BIT(0x605C)
+#define ACT_CTRL_0				AP1302_REG_16BIT(0x6066)
+#define ACT_CTRL_1				AP1302_REG_16BIT(0x6068)
 
 #define AP1302_V4L2_CTRLS			7
 #define AWB_MODE_0				0x00000000
@@ -2785,6 +2788,21 @@ static int ap1302_load_firmware(struct ap1302_device *ap1302)
 	if (ret)
 		return ret;
 
+	if (!strcmp(ap1302->sensor_info->model, AR1335_MODEL)) {
+		/* Set Actuator absolute milisecond time to 5ms */
+		ret = ap1302_write(ap1302, ACT_ABS_MTIME, 0x00001388, NULL);
+		if (ret)
+			return ret;
+		/* Initialise Actuator controls */
+		ret = ap1302_write(ap1302, ACT_CTRL_0, 0x00004C00, NULL);
+		if (ret)
+			return ret;
+
+		ret = ap1302_write(ap1302, ACT_CTRL_1, 0x00000110, NULL);
+		if (ret)
+			return ret;
+	}
+
 	/*
 	 *This is to check if PLL gets locked with new configurations
 	 */
-- 
2.17.1

